services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceDto.java

```diff
@@ -32,6 +32,12 @@
     @JsonProperty(value = RegistryManagementConstants.FIELD_PAYLOAD_DEVICE_ID)
     private String deviceId;
 
+    @JsonProperty(RegistryManagementConstants.FIELD_AUTO_PROVISIONED)
+    private Boolean autoProvisioned;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_AUTO_PROVISIONING_NOTIFICATION_SENT)
+    private Boolean autoProvisioningNotificationSent;
```

Comment:
is it important that these are not of the primitive type? IMHO the code will be easier to understand if we do not need to account for `null` values all over the place ...

Reply:
I think we could use the primitive type here. 

The only consequence as far as I can see would be: the autoProvisioningNotificationSent property would always be returned, also in cases of not-auto-provisioned devices it would also be returned as "false" (instead of being left out).

But I guess that's o.k. from a usability point of view?

Reply:
I see. Then how about letting the *isAutoProvisioned* and *isAutoProvisioningNotificationSent* methods return a primitive boolean instead?

Reply:
wouldn't it be easier to keep a *DeviceStatus* typed field instead of duplicating the *DeviceStatus* properties here?